proof/verifier.go

```diff
@@ -221,30 +238,203 @@ func (p *Proof) verifyExclusionProofs() (*commitment.TapCommitmentVersion,
 		return nil, ErrMissingExclusionProofs
 	}
 
-	// If there were no commitments in any exclusion proofs, then there is
-	// no version to return.
+	// If no commitments in proofs, no version to return.
 	if len(commitVersions) == 0 {
 		return nil, nil
 	}
 
-	// All ExclusionProofs must have similar versions.
-	firstCommitVersion := maps.Values(commitVersions)[0]
-	for outputIdx, commitVersion := range commitVersions {
-		outputCommitVersion := commitVersion
-		if !commitment.IsSimilarTapCommitmentVersion(
-			&firstCommitVersion, &outputCommitVersion,
-		) {
+	// All proofs must have similar versions.
+	firstVersion := maps.Values(commitVersions)[0][0]
```

Comment:
Verify this with an assertion here? `lfn.All`

Reply:
No, I don't think that's a particular good fit right here. Two reasons:
- `verifySTXOVersions` has two nested loops, so that would mean two nested `lfn.All`, which is kinda ugly.
- I want to log the exact proof that has the version that is different from the version of the first proof.

It's probably all possible with lfn.All, but it doesn't improve readability imo.